File: networktables/networktables.java
package org.usfirst.frc7108.Robot;
import edu.wpi.first.wpilibj.networktables.NetworkTable;

public class Robot extends TimedRobot 
{
	public static NetworkTable table;

    @Override
    public void robotInit() 
    {
    	table = NetworkTable.getTable("datatable");
    }
    @Override
	public void teleopPeriodic() 
	{
		int x = 1;
		// For co-processor to read
		table.putNumber("X",x);
		// 0.0 is default, when no data is available
		// Read data from co-processor
		int y = table.getNumber("Y", 0.0); 
	}

File: networktables/networktables.py
from networktables import NetworkTables
# To see messages from networktables, you must setup logging
import logging
logging.basicConfig(level=logging.DEBUG)

ip = "10.71.8.2" # ip of roborio
# Init networktables
NetworkTables.initialize(server=ip)
# Get the table
table = NetworkTables.getTable("datatable")
i = 0
while True:
	# Read data from the table
    print('X:', table.getNumber('X', 'N/A'))
    # Write data to table
    table.putNumber('Y', i)
    i += 1
